server/src/main/java/org/apache/iotdb/db/engine/storagegroup/TsFileResource.java

```diff
@@ -577,32 +570,62 @@ public int hashCode() {
     return Objects.hash(file);
   }
 
-  public void setClosed(boolean closed) {
-    this.closed = closed;
-  }
-
   public boolean isDeleted() {
-    return deleted;
-  }
-
-  public void setDeleted(boolean deleted) {
-    this.deleted = deleted;
+    return this.status == TsFileResourceStatus.DELETED;
   }
 
   public boolean isCompacting() {
-    return isCompacting;
-  }
-
-  public void setCompacting(boolean compacting) {
-    isCompacting = compacting;
+    return this.status == TsFileResourceStatus.COMPACTING;
   }
 
   public boolean isCompactionCandidate() {
-    return compactionCandidate;
+    return this.status == TsFileResourceStatus.COMPACTION_CANDIDATE;
   }
 
-  public void setCompactionCandidate(boolean compactionCandidate) {
-    this.compactionCandidate = compactionCandidate;
+  public boolean setStatus(TsFileResourceStatus status) {
+    boolean success = false;
+    switch (status) {
+      case CLOSED:
+        {
+          if (this.status != TsFileResourceStatus.DELETED) {
+            this.status = TsFileResourceStatus.CLOSED;
+            success = true;
+          }
+          break;
+        }
+      case UNCLOSED:
+        // Can we set a tsfile resource to unclosed status?
```

Comment:
Check code usages to confirm this. If no place uses this, print a warn (or error) log with stack trace here. Other cases should be processed the same (print a log when unexpected cases happen).

Reply:
We'd set it to an unclosed state in a test case in RewriteCompactionFileSelectorTest.  So I think we could just keep it and print a warning to users.

Reply:
resolve